### PR TITLE
Add hugo-preview-obsidian

### PR DESCRIPTION
**Pull request by fzdwx**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin:

https://github.com/fzdwx/hugo-preview-obsidian

## Release Checklist
- [ ] I have tested the plugin on
  - [ ]  Windows
  - [ ]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by liamcain** [comment]
- [isDesktopOnly](https://github.com/fzdwx/hugo-preview-obsidian/blob/5392fa69f410d31fe153883b6814232ee6a38eee/manifest.json#L9) you should mark this as true. child_process is only available on desktop
- [this.clean();](https://github.com/fzdwx/hugo-preview-obsidian/blob/5392fa69f410d31fe153883b6814232ee6a38eee/src/main.ts#L73-L74) The reason it's not quiting the process when Obsidian is closed is because unload isn't called on quit. If you want to run something when Obsidian gets quit, you should listen for the [`quit`](https://github.com/obsidianmd/obsidian-api/blob/38dd22168d2925086371bfc59e36fd9121527a39/obsidian.d.ts#L4266) event.

**Comment by fzdwx** [comment]
hello, is this how quit events are listened to:

```ts
this.registerEvent(this.app.workspace.on('quit', () => {
	this.clean();
}));
```

**Comment by liamcain** [comment]
If this.clean is synchronous, then yes, that should be all you need.

**Comment by fzdwx** [comment]
Ok, i released [0.1.2](https://github.com/fzdwx/hugo-preview-obsidian/releases/tag/0.1.2) and commit is https://github.com/fzdwx/hugo-preview-obsidian/commit/1131ecc4e893c4216d1e6375417eb8ce9199f279

File: community-plugins.json [hunk 1]
```diff
@@ -5913,6 +5913,13 @@
...
         "branch": "master"
-    }
+    },
+    {
+        "id": "hugo-preview-obsidian",
+        "name": "Hugo preview obsidian",
+        "author": "fzdwx",
+        "description": "hugo preview for obsidian",
+        "repo": "fzdwx/hugo-preview-obsidian",
+        "branch": "main"
```
Comment (liamcain) [inline community-plugins.json:5923]:
```suggestion
    {
        "id": "hugo-preview-obsidian",
        "name": "Hugo preview obsidian",
        "author": "fzdwx",
        "description": "hugo preview for obsidian",
        "repo": "fzdwx/hugo-preview-obsidian",
        "branch": "main"
    }
```

**Comment by fzdwx** [comment]
Thanks!

**Pull request merged**